core/trino-main/src/main/java/io/trino/sql/ir/Row.java

```diff
@@ -19,24 +19,32 @@
 import io.trino.spi.type.Type;
 
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 import static java.util.Objects.requireNonNull;
 
 @JsonSerialize
-public record Row(List<Expression> items)
+public record Row(List<Field> fields)
```

Comment:
It shouldn't be necessary to add the field names in the IR. As far as I recall, there's nothing in the planner/optimizer that can or needs to use them.